Fix unexpected key pixel_metrics.AUPRO.fpr_limit error

FanJiangIntel opened the pull request:
# Description

- Fix unexpected key error of pixel_metrics.AUPRO.fpr_limit

- Fixes #988

## Changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which refactors the code base)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

- [ ] My code follows the [pre-commit style and check guidelines](https://github.com/openvinotoolkit/anomalib/blob/main/CONTRIBUTING.md) of this project.
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing tests pass locally with my changes
- [ ] I have added a summary of my changes to the [CHANGELOG](https://github.com/openvinotoolkit/anomalib/blob/main/CHANGELOG.md) (not for minor changes, docs and tests).

FanJiangIntel committed:
fix unexpected key pixel_metrics.AUPRO.fpr_limit

Signed-off-by: FanJiangIntel <fan.jiang@intel.com>

WenjingKangIntel commented on src/anomalib/utils/metrics/aupro.py:
```diff
@@ -48,7 +48,7 @@ def __init__(
         self.add_state("preds", default=[], dist_reduce_fx="cat")  # pylint: disable=not-callable
         self.add_state("target", default=[], dist_reduce_fx="cat")  # pylint: disable=not-callable
-        self.register_buffer("fpr_limit", torch.tensor(fpr_limit))
+        self.register_buffer("fpr_limit", torch.tensor(fpr_limit), persistent=False)
```
Please help double check whether removing it from the `state_dict` is reasonable, especially when we want to use it during testing.

FanJiangIntel replied:
Thanks! I just pushed new commit to keep loading `fpr_limit` from `state_dict` by creating `AUPRO` instance before loading model, please take a look.

FanJiangIntel committed:
load AUPRO before create_metric_collection

Signed-off-by: FanJiangIntel <fan.jiang@intel.com>

WenjingKangIntel commented on src/anomalib/models/components/base/anomaly_module.py:
```diff
@@ -241,4 +242,9 @@ def load_state_dict(self, state_dict: OrderedDict[str, Tensor], strict: bool = T
         # Used to load missing normalization and threshold parameters
         self._load_normalization_class(state_dict)
+        # Used to load fpr_limit in AURPO before create_metric_collection
+        if not hasattr(self, "pixel_metrics") and "pixel_metrics.AUPRO.fpr_limit" in state_dict.keys():
```
Please put it in a new function like `self._load_normalization_class`.

FanJiangIntel replied:
Have fixed!

FanJiangIntel committed:
code refine

Signed-off-by: FanJiangIntel <fan.jiang@intel.com>

FanJiangIntel committed:
fix comment

Signed-off-by: FanJiangIntel <fan.jiang@intel.com>

FanJiangIntel committed:
fix

Signed-off-by: FanJiangIntel <fan.jiang@intel.com>

WenjingKangIntel committed:
Support test

Signed-off-by: Kang Wenjing <wenjing.kang@intel.com>

samet-akcay committed:
Merge branch 'main' into bug-988-dev

WenjingKangIntel committed:
Update test

Signed-off-by: Kang Wenjing <wenjing.kang@intel.com>

WenjingKangIntel committed:
Update test

Signed-off-by: Kang Wenjing <wenjing.kang@intel.com>

WenjingKangIntel committed:
🛠 Fix mypy pep561 (#1088)

Signed-off-by: Kang Wenjing <wenjing.kang@intel.com>

samet-akcay committed:
📝 [Notebooks] - Simplify the dobot notebooks. (#1084)

* Fix metadata path

* Modified 501b to use API

* Add 501a notebook

* Renamed the notebooks

* update the readme file with the new notebook name

* Use anomalibs download_and_extract function to download the dataset

* Use anomalibs download_and_extract function to download the dataset

* Cleanup cell outputs

* Simplify the inference notebook

* Address PR comments

* Fix pre-commit

* Updating new notebooks names

* Updating new notebooks names

* Update the readme file

The previous readme file had some errors. It was mentioned NNCF. We must also include the step to create the environment and install Anomalib and jupyter notebooks. Anomalib installation will also be in the notebook, and that is fine.

* fixing broken link

---------

Co-authored-by: Paula Ramos <paula.ramos@intel.com>

samet-akcay committed:
📝 [Notebooks] Fix the broken formatting (#1090)

* Fix metadata path

* Format the notebook

ashwinvaidya17 committed:
🧪 Add tests for tools (#1069)

* 🧪 Add tests for tools

* Add dev.txt

* Add type:ignore

* Add pytest-order plugin to tox

* Address PR comments + codacy issues

* Remove setup

* Address codacy issue

---------

Co-authored-by: Ashwin Vaidya <ashwinitinvaidya@gmail.com>

samet-akcay committed:
Upgrade python to 3.10 (#1035)

* Fix metadata path

* Upgrade python to 3.10

* update pre-commit

* Update docs.yml

* Update pre_merge.yml

* Fix python version in pre-merge.yaml

* Removed cuda10.2 dockerfile and renamed cuda11.4 as Dockerfile

* Temporarily reduce the coverage

Ravindu987 committed:
Add kolektor dataset (#983)

* Add dataset create function

* Add KolektorDataset class

* Add Kolektor class

* Add comments for dataset creation

* Add licensing

* Fix error message

* Add train split ratio as a parameter

* Add dataset link

* Add docstrings

* Fix warn in imageio

* Add Prepare dataset method with custom download function

* Code clean

* Update function to check mask

* Update download util with filename in DownloadInfo

* Use updated download_and_extract in Kolektor

* Use opencv for reading images

* Updated the changelog

---------

Co-authored-by: Samet Akcay <samet.akcay@intel.com>

samet-akcay committed:
📝 [Notebooks] - Install anomalib via pip in the Jupyter Notebooks (#1091)

* Fix metadata path

* Refactor btech notebook to remove clone stuff

* Refactor datamodule notebooks

* Update fastflow model api

* Add dataset symlink to notebooks

* Fix pre-commit

yunchu committed:
Update code-scan workflow to use Trivy (#1097)

* update code-scan workflow to use Trivy

* Update tox.ini

Co-authored-by: Ashwin Vaidya <ashwin.vaidya@intel.com>

* Update .github/workflows/code_scan.yml

Co-authored-by: Ashwin Vaidya <ashwin.vaidya@intel.com>

---------

Co-authored-by: Ashwin Vaidya <ashwin.vaidya@intel.com>

ashwinvaidya17 committed:
Patch release 0.5.1 (#1101)

* Set init_state_dict

* pass openvino metadata as config

* DictConfig -> dict

* Update changelog

* Update date

* metadata_path -> metadata

---------

Co-authored-by: Ashwin Vaidya <ashwinitinvaidya@gmail.com>

dependabot[bot] committed:
Bump requests from 2.26.0 to 2.31.0 in /requirements (#1100)

* Bump requests from 2.26.0 to 2.31.0 in /requirements

Bumps [requests](https://github.com/psf/requests) from 2.26.0 to 2.31.0.
- [Release notes](https://github.com/psf/requests/releases)
- [Changelog](https://github.com/psf/requests/blob/main/HISTORY.md)
- [Commits](https://github.com/psf/requests/compare/v2.26.0...v2.31.0)

---
updated-dependencies:
- dependency-name: requests
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

* Update openvino.txt

---------

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Samet Akcay <samet.akcay@intel.com>

ashwinvaidya17 committed:
Update main to dev (#1104)

update main to dev

Co-authored-by: Ashwin Vaidya <ashwinitinvaidya@gmail.com>

ashwinvaidya17 committed:
🗝️ Pass codedov token from environment (#1102)

* with secrets test

* Use action

* modify touch command

* Change proxy url

* Use token from env

* missing download

* Modify path

* Restore workflow

* Update README

---------

Co-authored-by: Ashwin Vaidya <ashwinitinvaidya@gmail.com>

samet-akcay committed:
Add dataset categories to data modules (#1105)

* Fix metadata path

* Add categories to data modules

xbkaishui committed:
fix tile import typo (#1106)

fix import typo

fix example code.
change 
from anomalib.data.tiler import Tiler
to
from anomalib.pre_processing import Tiler

Commit:
OV API2.0 (#1098)

* OV API2.0

* NNCF error file allocation

* OV_API_2.0_updated

* Update src/anomalib/deploy/inferencers/openvino_inferencer.py

* Address pre-commit issues

* Remove redundant lines

---------

Co-authored-by: Ashwin Vaidya <ashwinnitinvaidya@gmail.com>
Co-authored-by: Samet <samet.akcay@intel.com>

jasonvanzelm committed:
Fix Fastflow ONNX export. (#1108)

* Fix Fastflow ONNX export.

* Refine code comment

Co-authored-by: Samet Akcay <samet.akcay@intel.com>

---------

Co-authored-by: jasonvanzelm <vanzelm@gestalt-robotics.com>
Co-authored-by: Samet Akcay <samet.akcay@intel.com>

djdameln committed:
AI-VAD bbox post-processing (#1103)

* add bbox postprocessing methods (WIP)

* delete overlapping bboxes

* add device handling in box utils

* add foreground detections

* remove unused method

* add docstrings

* expose region extractor parameters

* update changelog

* update changelog

* address PR comments

alexriedel1 committed:
EfficientAD (#1073)

* reading augmentations from config

* missing if

* remove debug

* pre-commit hooks

* solve memory cumulation

* fixing imports

* uncomment pre commits

* EfficientAD first commit

* fix pre commit

* download imagenet subset

* image size and channel out

* image size and channel out

* remove saved statistics

* autoencoder out channels

* config image size

* gdrive download path to string

* hash for imagenet subset

* efficientad config

* reload dataloader every epoch

* calculate teacher channel mean only on train start

* set channel and quantiles as parameter

* teacher normalization in forward

* quantiles tensor 0.0

* on train end quantiles

* new mode name

* on train end quantiles

* eval after train

* 70k steps

* save before validation

* manual get last epoch

* set to train mode

* remove quantiles from step

* on validation start calculate quantiles

* remove eval

* logger info

* logger and no desc

* set to last epoch

* training epoch end

* tqdm position

* training epoch end args

* eval model

* delete outputs

* on train end

* on training epoch end

* check teacher model path

* other maps calculation?

* on train epoch end

* remove print in tools

* on train epoch end

* doctstrings

* quantiles on validation start

* quantiles only on last epoch

* fix private

* pre-commit

* docstrings

* grad on teacher model, pin memory

* alternative teacher

* loss calculation

* loss calculation

* moved imagenet normalition to models

* pre commit hooks

* transform

* transform line

* pre commit hooks

* new teacher, simpler architecture

* map normalization with oinly good images

* pre-commit

* loss weighting

* all pretrained models

* added small pretrained

* lr scheduler

* pre commit

* remove loss weighting

* 200 epochs

* num steps

* get training steps in on train start

* get nuim steps in optimizer

* loss weighting

* no loss weighting

* remove normalization epsilon

* torchvision transforms for imagenet

* albumentations imagenet

* adamw

* add classification label to segmentation task

* undo visualizer refactor

* refactor inference

* pre commit hooks

* compute distance_st only once

* torchvision imagenet

* albumentation resize method

* best settings

* access dict

* ordering...

* model and padding true

* pre-commit

* add gdown

* solve conflict

* list kwarg and random choice

* imagenette

* imagenette hash

* validation set for feature map normalizatioN

* remove gdown

* num epochs

* feature maps from train images

* minor fixes

* new pretrained teachers

* no teachers

* new teachers

* remove weights, add weight download, add support for arbitrary image sizes

* remove pretrained folder from git

* remove pretrained folder from git

samet-akcay committed:
Fix `pre-commit` issues caused by the EfficientAD PR (#1114)

* Fix metadata path

* Run pre-commit on README.md

* Add efficientad to tests

* Add img_size to Decoder

* Reduce the coverage to 70%

* temporarily remove efficientad

samet-akcay committed:
Release `v0.6.0` (#1133)

* Fix metadata path

* Release v0.6.0

* Address pre-commit issues

samet-akcay committed:
Version bump (#1135)

* Fix metadata path

* Bump version

yunchu committed:
added a test workflow to stick to the pr-check workflow completed event (#1157)

I'm not sure there would be another solution to test this but the workflow which will be triggered by 'workflow_run' event should be in the default branch so I need to add it to the main for testing.

ORippler committed:
Fix typechecking error for toch.onnx.export (#1159)

nelson1425 committed:
Update EfficientAD (#1143)

* Update EfficientAD

* Update src/anomalib/models/efficientad/lightning_model.py

Co-authored-by: Declan <dkutscher10@gmail.com>

* Update src/anomalib/models/efficientad/lightning_model.py

Co-authored-by: Declan <dkutscher10@gmail.com>

* Replace 16777216 with 2**24

* Add pad_maps config parameter

---------

Co-authored-by: Declan <dkutscher10@gmail.com>

blaz-r committed:
Fix benchmark (#1155)

* Fixed broken links in readme

* Fixed inference command in readme

* Fix benchmark torch throughput

* Add tests to cover throughput

* Add fixture path to conftest

* Format changed files

* Add implementation to conftest

samet-akcay committed:
🚜 Refactor: Add `get_available_models()` support. (#1166)

* Fix metadata path

* Renamed EfficientAD to EfficientAd

* Add backwards compatibility for efficientad name

* Add get_available_models() support

samet-akcay committed:
Add default values to algorithms  (#1169)

* Fix metadata path

* Add default values to Cfa

* Add default values to Cflow

* Add default values to CsFlow

* Add default values to DFKDE

* Add default values to DFM

* Add default values to EfficientAd

* Add default values to FastFlow

* Add default values to Ganomaly

* Add default values to Padim

* Add default values to PatchCore

* Add default values to ReverseDistillation

* Add default values to STFPM

* Update the changelog

samet-akcay committed:
Revert "Add default values to algorithms " (#1193)

Revert "Add default values to algorithms  (#1169)"

This reverts commit 27876c851eb97a5b081b89a0059b7cbb2cad9688.

alexriedel1 committed:
Minor fixes (#1182)

* squeeze output map

* add resize from dataset config, openvino warning

* fix precommit

* OpenVino log warning

* pre commit

* squeeze maps and mask

* quantile element reduction

* docstring, fixes

* f strings

---------

Co-authored-by: Samet Akcay <samet.akcay@intel.com>

samet-akcay committed:
🚜 Address bandit issues (#1152)

* Fix metadata path

* Address bandit issues

* Address codacy issues

* Changed the bandit configuration file

* Address PR comments

* bandit fix.

samet-akcay committed:
Configure readthedocs via `.readthedocs.yaml` file (#1229)

ashwinvaidya17 committed:
🚚 Refactor Benchmarking Script (#1216)

* New printing stuff

* Remove dead code + address codacy issues

* Refactor try/except + log to comet/wandb during runs

* pre-commit error

* third-party configuration

---------

Co-authored-by: Ashwin Vaidya <ashwinitinvaidya@gmail.com>

samet-akcay committed:
Update CODEOWNERS

djdameln committed:
Enable training with only normal images for MVTec (#1241)

* ignore mask check when dataset has only normal samples

* update changelog

samet-akcay committed:
Revert "🚚 Refactor Benchmarking Script" (#1239)

Revert "🚚 Refactor Benchmarking Script (#1216)"

This reverts commit 784767fc2f19a8f354f152aba7f4338cb628118c.

samet-akcay committed:
Merge branch 'main' into bug-988-dev

samet-akcay committed:
Update benchmarking notebook (#1242)

* Fix metadata path

* Update benchmarking notebook

blaz-r committed:
Fix links to model architecture images (#1245)

* Fix links to architecture images

* Change links to raw files

Seanny123 committed:
Wandb unwatch method belongs to experiment, not logger (#1246)

unwatch method belongs to experiment, not logger

JoaoGuibs committed:
(Minor change) Added the tracer_kwargs to the TorchFXFeatureExtractor class (#1214)

* Added tracer_kwargs to torchfx

* Added tracer_kwargs on docstring

blaz-r committed:
Replace cdist in Patchcore (#1267)

samet-akcay committed:
Ignore hidden directories when creating `Folder` dataset (#1268)

samet-akcay committed:
Remove `config` from argparse in OpenVINO inference script. (#1257)

* Fix metadata path

* Remove leftover argument

* Update openvino entrypoint script

samet-akcay committed:
Fix EfficientAD number of steps for optimizer lr change? (#1266)

* Fix metadata path

* Fix number of steps

samet-akcay committed:
Add Details to the PR Template (#1269)

* Fix metadata path

* Ignore hidden directories in folder dataset

* Update pr template

* Update pull_request_template.md

* Update pull_request_template.md

Commit:
Add support for receiving folder dataset paths as a list (#1265)

* Add List input as folder path

* Modify CHANGELOG.md

* Update if not to if

* Update ListConfig to Sequence

* Add test case for folder dataset with list input.

* Rename function & reflect reviews

* Change test dataset bottle to hazelnut_toy

* Change docstring

* Fix test function docstring

* Recheck test dataset paths

* mask_dir to None

---------

Co-authored-by: Samet Akcay <samet.akcay@intel.com>

ashwinvaidya17 committed:
🚨 Test hotfix (#1280)

Fix test

samet-akcay committed:
Check if `mask_dir` is available for segmentation task in `Folder` data (#1277)

* Fix metadata path

* Ignore hidden directories in folder dataset

* Add check for mask_dir for segmentation tasks in Folder dataset

* Check the test split mode when no mask dir is provided

ashwinvaidya17 committed:
Add modelAPI related stuff (#1219)

* Add modelAPI related stuff

* Address PR comments + fix tests

* Add comment + modify export

* Update src/anomalib/deploy/export.py

Co-authored-by: Samet Akcay <samet.akcay@intel.com>

---------

Co-authored-by: Ashwin Vaidya <ashwinitinvaidya@gmail.com>
Co-authored-by: Samet Akcay <samet.akcay@intel.com>

Yann-CV committed:
Add binning capability to AUPRO (#1145)

* Add the capability to compute binned AUPRO.

* fix linting

* use directly binary_roc

* update CHANGELOG.md

* improve test by doing 2 different ones (aupro and binned aupro) + renamed few variables in the tests

* only allow num_thresholds as input + fix tests + add threshold computing utilities

* add binning tests

* use binary roc directly

* remove unused import and rename some

* device for thresholds

* fix linting

* use torch.all

* fix linting

* fix linting

* remove observe time

---------

Co-authored-by: Samet Akcay <samet.akcay@intel.com>

alexriedel1 committed:
Add status progress bar to patchcore coreset sampling (#1289)

* squeeze output map

* add resize from dataset config, openvino warning

* fix precommit

* OpenVino log warning

* pre commit

* squeeze maps and mask

* openvino import and tqdm coreset

* spelling

---------

Co-authored-by: Samet Akcay <samet.akcay@intel.com>

ashwinvaidya17 committed:
Release `v0.7.0` (#1304)

ashwinvaidya17 committed:
Modify workflow (#1306)

Kiminjo committed:
[Chore] Modify README custom dataset  (#1314)

* [Chore] Add several key values to the custom dataset on README.md

* [Chore] Add val_split_ratio key values to the custom dataset on README.md

ashwinvaidya17 committed:
Version bump (#1305)

Bump version

Co-authored-by: Ashwin Vaidya <ashwinitinvaidya@gmail.com>
Co-authored-by: Samet Akcay <samet.akcay@intel.com>

ashwinvaidya17 committed:
🩹 Fix docs build (#1307)

* Modify workflow

* use self hosted runner

* quotes python version

* pre-commit whitespace

* revert cp stage

---------

Co-authored-by: Ashwin Vaidya <ashwinitinvaidya@gmail.com>
Co-authored-by: Samet Akcay <samet.akcay@intel.com>

samet-akcay committed:
Fix the readthedocs config (#1322)

* Fix metadata path

* Ignore hidden directories in folder dataset

* Add check for mask_dir for segmentation tasks in Folder dataset

* Fix the readthedocs config

* Remove hadolint + install python package

---------

Co-authored-by: Ashwin Vaidya <ashwinitinvaidya@gmail.com>

blaz-r committed:
Fix PRO metric calculation on GPU (#1317)

* Fix shapes and types in pro metric

* Adjust pro metric tests to fit real scenario

* Fix formatting in test

* Refactor target reshaping

Co-authored-by: Samet Akcay <samet.akcay@intel.com>

---------

Co-authored-by: Samet Akcay <samet.akcay@intel.com>

samet-akcay committed:
Change the documentation URL in `README.md` and add commands to run each inferencer (#1326)

* Fix metadata path

* Ignore hidden directories in folder dataset

* Add check for mask_dir for segmentation tasks in Folder dataset

* Replace docs

* Add each inferencing scripts as a details section

* Update README.md

* Update README.md

phcarval committed:
Fix dockerfile cuda version (#1330)

fixed dockerfile cuda version

murilo-cunha committed:
Add `setuptools` as a requirement (via `pkg_resources`) (#1168)

Usually, this comes with `pip`, but it's possible to have a virtual environment without pip installed.

aadhamm committed:
adding KMeans PyTorch Implementation to cfa model (#998)

* adding KMeans PyTorch Implementation to cfa model

* Required modifications and implementation comparison

* Update torch_model.py

* Update torch_model.py

* Update kmeans.py

* required changes and Google docstring format

* remove test notebook

* Update torch_model.py

* 'unindent does not match any outer indentation level (<unknown>, line 34)' (syntax-error)

* Update torch_model.py

* Fix pre-commit issues

* Revert torch model to sklearn

---------

Co-authored-by: Samet Akcay <samet.akcay@intel.com>
Co-authored-by: Ashwin Vaidya <ashwin.vaidya@intel.com>

jpcbertoldo committed:
Fix patchcore interpolation (#1335)

* correct interpolation

* deal with edge case

MG109 committed:
Efficient ad reduced memory footprint (#1340)

* implement teacher_channel_mean_std with lower mem. footprint

Workaround loops over dataset twice: no need to save teacher_outputs (which needs much GPU memory for large datasets)

* Fix pre-commit

* implement new mean/std calculation as standard

---------

Co-authored-by: Samet <samet.akcay@intel.com>

BeeAlarmed committed:
Allow dynamic batch-sizes when exporting to ONNX, instead if fixed input shapes (#1347)

* Allow dynamic axis when exporting to ONNX, instead if fixed input

* Update src/anomalib/deploy/export.py

---------

Co-authored-by: Fabian <fabian.hickert@posteo.de>
Co-authored-by: Samet Akcay <samet.akcay@intel.com>

mzweilin committed:
Add support to backbone URI in config. (#1343)

pirnerjonas committed:
fix(docs): typo (#1353)

samet-akcay committed:
Add extra checks to `TorchInferencer` model and metadata loading (#1350)

* Fix metadata path

* Ignore hidden directories in folder dataset

* Add check for mask_dir for segmentation tasks in Folder dataset

* Add extra checks to ensure that torch model has metadata and model information

blaz-r committed:
Fix tiling for Reverse Distillation and STFPM (#1319)

* Fix tiling for stfpm

* Fix tiling for reverse distillation

* Add tiling to fastflow

* Make tiling in fastflow backwards compatible

* Rename image_size to anomaly_map_size.

* Reverse fastflow tiling

* Remove tiling calls in fastflow

---------

Co-authored-by: Samet Akcay <samet.akcay@intel.com>

sovrasov committed:
README: Synchronize OV version with requirements (#1358)

samet-akcay committed:
Update the broken link in training.rst (#1363)

abc-125 committed:
Fix EfficientAD to use width and height of the input (#1355)

* Fix EfficientAD to use two dimensions of the input

* Update src/anomalib/models/efficient_ad/torch_model.py

Co-authored-by: Samet Akcay <samet.akcay@intel.com>

* Update torch_model.py

---------

Co-authored-by: Samet Akcay <samet.akcay@intel.com>

Gornoka committed:
chore(deps): update timm to 0.6.13 (#1373)

updates the timm version to 0.6.13, this includes a fix for python3.11 related dataclass problems.

samet-akcay committed:
Handle `dict` objects in `TorchInferencer` (#1354)

* Fix metadata path

* Ignore hidden directories in folder dataset

* Add check for mask_dir for segmentation tasks in Folder dataset

* Torch Inferencer handle dict output

* Rename pred_score anomaly_score to pred_score

* Fix tests

* Compute pred_score from anomaly_map if it is not available in dict

* Rename anomaly_map_combined to anomaly_map in efficient_ad

caiolang committed:
Missing indentation in metrics.rst docs (#1379)

`compute_on_cpu` should be an internal parameter of `init_args` in the `metrics` configurations.

ashwinvaidya17 committed:
Add tag to workflow (#1382)

Add anomalib tag to runner + action yaml

---------

Co-authored-by: Ashwin Vaidya <ashwinitinvaidya@gmail.com>

djdameln committed:
Address tiler issues (#1411)

* fix tiler

* deprecate random tile locations

* restore random tiling in tile method

* check tiling section in config

* disable tiling for ganomalu

* pad -> padding

abc-125 committed:
Refactor Reverse Distillation to match official code (#1389)

* Non-mandatory early stopping

* Added conv4 and bn4 to OCBE

* Loss as in the official code (flattened arrays)

* Added comment on how to use torchvision model as an encoder to reproduce results in the paper

* Remove early stop from config, change default anomaly_map_mode to add

* pre-commit fix

* Updated results

* Update src/anomalib/models/reverse_distillation/README.md

Co-authored-by: Samet Akcay <samet.akcay@intel.com>

* Update src/anomalib/models/reverse_distillation/README.md

Co-authored-by: Samet Akcay <samet.akcay@intel.com>

* Update src/anomalib/models/reverse_distillation/README.md

Co-authored-by: Samet Akcay <samet.akcay@intel.com>

* Remove early_stopping

* Update src/anomalib/models/reverse_distillation/lightning_model.py

Co-authored-by: Samet Akcay <samet.akcay@intel.com>

* Easier to read code

---------

Co-authored-by: Samet Akcay <samet.akcay@intel.com>

ggiret-thinkdeep committed:
Patch for the WinError183 on the OpenVino export mode (#1386)

* Fix WinError183 (Windows Error)

* Add commentary of the change

---------

Co-authored-by: Youho99 <gaylord.giret@viacesi.fr>

samet-akcay committed:
Merge branch 'main' into bug-988-dev
